### ISSUE TITLE
Line and Area charts not working when Replace Missing Values not set to Linear Interpolated and server timezone is set to UTC

### ISSUE THREAD

**Issue by bendog** [metabase/metabase#4802]
Hey,

I've noticed an issue recently, and I'm not sure if it's related to a version update of Metabase or perhaps MySQL, or not, I don't recall doing anything specific which might have caused this issue. 

 it seems for some reason the max and sum values returned from my query are being interpreted as null values by the line and area charts. 

This only seems to effect charts where Replace Missing Values is set to Zero or Nothing. 
Linear interpolated seems to work fine. 

Tried across multiple browsers, both on Mac OS and iPad 
Host OS: ubuntu 16.04
Base DB for Metabase is MySQL 5.6
Reporting data DB is MySQL 5.6
Metabase version: 0.22.2 .jar
Also tested with 0.23.1 .jar

![img_1136](https://cloud.githubusercontent.com/assets/894916/25114094/9b3c114a-242e-11e7-823c-7ceb7cdf3274.PNG)
![img_1137](https://cloud.githubusercontent.com/assets/894916/25114095/9b9b970a-242e-11e7-8aa8-eec7bc2c6f26.PNG)
![img_1138](https://cloud.githubusercontent.com/assets/894916/25114096/9ba46696-242e-11e7-8df2-eb0ba8f6d7b5.PNG)

**Comment by bendog** [comment]
This seems related to #4277, however this issue is still occurring after upgrade to latest version?

**Comment by mazameli** [comment]
I was able to reproduce this.

**Comment by mfpinhal** [comment]
This is happening to us as well in many different cases. Another example:

Replace missing values with: **Linear interpolated**:
![image](https://cloud.githubusercontent.com/assets/2720934/26752372/30bdecda-4846-11e7-932a-1127f16ba6e6.png)

Replace missing values with: **Zero**:
![image](https://cloud.githubusercontent.com/assets/2720934/26752388/9926e164-4846-11e7-8657-ef624c79af2e.png)

**Comment by bendog** [comment]
@mfpinhal it looks like that is working as expected, as you are getting one result of 1.0 and the rest are 0?
can you show the raw tables?

**Comment by mfpinhal** [comment]
@bendog, indeed the raw tables return values! The raw table that feeds the charts above:

![image](https://cloud.githubusercontent.com/assets/2720934/26829701/6aff0aee-4abe-11e7-8c4c-e35503b43ba3.png)

**Comment by bendog** [comment]
that looks very unusual, so there should be results, however only the first one is showing?

**Comment by vitorbaptista** [comment]
I'm seeing the same issue on Metabase v0.24.1

**Comment by mfpinhal** [comment]
> that looks very unusual, so there should be results, however only the first one is showing?

This happens in many many cases, can we prioritize this one up?

Replace missing values with: **Linear interpolated**:
<img src="https://user-images.githubusercontent.com/2720934/27488203-1fd05762-582e-11e7-9819-ce761f2e8a37.png" width="700">

Replace missing values with: **Zero**:
<img src="https://user-images.githubusercontent.com/2720934/27488241-3ec4ccb6-582e-11e7-8fcf-e455c81cf051.png" width="700">

**Comment by jrpeck1989** [comment]
I have this issue on 0.24.2 as well

**Comment by doodlebro** [comment]
This have priority? Seems like a pretty large oversight.

**Comment by ricardojmendez** [comment]
Confirmed that this is happening with:

- A line graph showing the count of rows by date, where not every date has a value;
- Data coming from Postgresql (although it looks to be purely a visualization bug);
- Metabase v0.24.2

I've set it to _show marker points_ to evaluate how it acts. The behavior is:

- The data is displayed correctly as long as it is Linear Interpolated;
- If I switch it to replace missing values with nothing, the graph is empty;
- If I set it to replace the missing values with zero, _every day will have a zero value - even those days that did not originally include a data point_.

**Comment by attekei** [comment]
I was able to reproduce this reliably on our hosted test Metabase instance but couldn't reproduce it locally in my development environment even though both instances run the same Metabase version, both are backed by PostgreSQL database and both access the same PostgreSQL dataset (our internal `Pies` test dataset). I also checked the response of `/api/dataset` but couldn't find differences that explain the different behavior.

I wonder if this has something to with settings or timezones or if I simply missed something important.

**Comment by camsaul** [comment]
I'm also not able to reproduce this locally or on our hosted test instance.

In fact, both "zero" and "linear interpolation" both appear to be doing the same thing for me, which is filling all then `NULL` values with zeroes. I used the following SQL to generate some data to test this (Postgres):

```sql
SELECT '2017-10-01'::date AS d, 3 AS c
UNION
SELECT '2017-10-02'::date AS d, 4 AS c
UNION
SELECT '2017-10-03'::date AS d, NULL AS c
UNION
SELECT '2017-10-04'::date AS d, 2 AS c
UNION
SELECT '2017-10-05'::date AS d, 4 AS c
UNION
SELECT '2017-10-06'::date AS d, NULL AS c
UNION
SELECT '2017-10-07'::date AS d, 9 AS c
UNION
SELECT '2017-10-08'::date AS d, 6 AS c
```

Can anyone here confirm this issue is still happening on Metabase >= 0.26.1 and, if so, give me step-by-step instructions for how to repro? SQL for some fake data like what I posted would be greatly appreciated.

**Comment by camsaul** [comment]
Anyone? Bump

**Comment by mfpinhal** [comment]
> Anyone? Bump

Still happenning on 0.26.1:

Linear interpolated:
![image](https://user-images.githubusercontent.com/2720934/32070466-5887b20e-ba84-11e7-9fee-f163f33e7c3d.png)

Replace missing values with Zero:
![image](https://user-images.githubusercontent.com/2720934/32070449-45730358-ba84-11e7-97e6-96b2d9f9f102.png)

**Comment by camsaul** [comment]
@mfpinhal can you show me what that data looks like as a table? Just a few rows or so should do

**Comment by mfpinhal** [comment]
@camsaul, looking deeper, I think I found something else, notice the dates on the chart, they don't match the values displayed, and same **weeks** label are repeated:

Replace with 0's:
![image](https://user-images.githubusercontent.com/2720934/32134247-1aecac66-bbe1-11e7-81c9-d28afa979ee1.png)
![image](https://user-images.githubusercontent.com/2720934/32134232-bad37544-bbe0-11e7-871f-ee8436ea554b.png)
![image](https://user-images.githubusercontent.com/2720934/32134246-01af4664-bbe1-11e7-964e-734736cde2c3.png)

And now, the table values, perfectly normal:
![image](https://user-images.githubusercontent.com/2720934/32134222-60b12764-bbe0-11e7-84ca-01710079ff41.png)

Although the same week dates/labels are displayed, with linear interpolated option, values come through ok though:
![image](https://user-images.githubusercontent.com/2720934/32134258-5fceb144-bbe1-11e7-8738-0d1606142af1.png)

Anyway, I hope this points you in the right direction to understand and fix this bug.

**Comment by mfpinhal** [comment]
Probably related to https://github.com/metabase/metabase/issues/5916 (actually, both cases refer to the same Metabase instance).

**Comment by arasmussen** [comment]
Hmm, I think I'm still seeing this issue in the 0.30 RC.

![image](https://user-images.githubusercontent.com/969540/43799440-69ad25de-9a42-11e8-8374-cfb7560fbe51.png)
![image](https://user-images.githubusercontent.com/969540/43799450-71fe08a2-9a42-11e8-8bec-bc736816a4e7.png)

cc @tlrobinson

**Comment by rylonian** [comment]
I'm still seeing this issue in V0.30.4.
I have tried multiple browsers and the results are the same.  Linear Interpolated works fine, but using zero or nothing sets all the x axis points in the graph to 0 or undefined.

Host OS: ubuntu 18.04
Base DB for Metabase is MySQL 5.7.23
Reporting data DB is MS SQL Server 2014 - 12.0.2269.0
Metabase version: 0.30.4.jar

Strangely, I have tested metabase on my local Windows 10 PC and it works fine there.  My windows PC just has Java 8 and I used the database created by metabase.  I hope this helps.

Thanks!

![image](https://user-images.githubusercontent.com/3903053/46301020-235cdf00-c573-11e8-9daf-92a410e7c63d.png)

![image](https://user-images.githubusercontent.com/3903053/46301040-32dc2800-c573-11e8-9584-bab590cd26ef.png)

**Comment by rylonian** [comment]
I did some more testing for this specific issue.  I have tested it on Ubuntu 18.04, Windows 10, and Server 2012 R2.  Ubuntu is the only operating system where changing the graph to use "Zero" or "Nothing" instead of "Linear Interpolated" causes the X axis to break.  I don't know why Ubuntu would work differently than Windows, but there is something.  

I have tested this with my personal data as well as the default sample data and I still achieve the same results.

To reproduce:
1. Install a fresh instance of metabase on Ubuntu and run it.
2. Setup account / whatever is needed just for the default sample data.
3. Ask a Question
4. Choose Custom with the following options:
Data: Orders
Filtered By: Created At Past 30 Days
View: Sum of Total
Grouped By: Created At: Day
5. Click Get Answer
6. Change the Visualization to a Line graph
7. Click the Gear icon -> Display and change the Replace missing values with option to Zero.

I was hoping to be able to run this on a Linux VM instead of Windows.  If anyone has a work around, please let me know.  I will be happy to test or provide whatever information is needed.

Thanks again!

**Comment by rylonian** [comment]
Final Comment for now.
This seems to be related to the server being set to the UTC timezone.  Once I changed the timezone on the server to my local timezone and restarted Metabase, the graphs now work as expected.

Thanks again!

**Comment by mfpinhal** [comment]
rylonian commented:
> This seems to be related to the server being set to the UTC timezone.

Indeed, I confirm my instance is set to the UTC timezone.

**Comment by listentojohan** [comment]
I just encountered the same issue. (utc timestamp fields)

**Comment by ValdarT** [comment]
Still there on v0.31.0. Highly problematic. :(

**Comment by keshunchen** [comment]
I'm still seeing this issue in V0.31.1

**Comment by sthornock** [comment]
So I think it is related to this..
If my x-axis is  a "date/time" then it only interpolated values work for missing values.
If I change the x-axis column to be something like this...in SQL server
convert(varchar(25), [Date], 101) As PlanDate then it works.

Try converting your x-axis column to a varchar and format it as a work around.

**Comment by tuliosciotta** [comment]
I'm still seeing this issue in V0.31.2...

**Comment by mazameli** [comment]
Bumping up the priority of this given how many people have run into it.

**Comment by bbqqcc6888** [comment]
When the Linear Interpolated  is selected, empty is still displayed on the chart
![image](https://user-images.githubusercontent.com/43432506/73339299-6d62eb80-42b3-11ea-9a6f-2e08a74f1c2d.png)

![image](https://user-images.githubusercontent.com/43432506/73339343-84a1d900-42b3-11ea-8f52-66fe58544748.png)


Host OS: Centos 7.6（Docker）
Base DB for Metabase is MySQL 8.0
Reporting data DB is MySQL 8.0
Metabase version: 0.34.1 .jar

**Comment by flamber** [comment]
I cannot reproduce this on 0.35.3 by following the steps in https://github.com/metabase/metabase/issues/4802#issuecomment-426017298 with `user.timezone=UTC`.
Think it was fixed in 0.34 along with many other timezone fixes.
Related #4122 - nulls are displayed as zero

**Comment by flamber** [comment]
@danimasa Please open a new issue and provide all the details and steps-to-reproduce, and which version it worked in.
